src/utils/dataMatchesContraints.js

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+module.exports = function(data, constraints) {
```

Comment:
Unless this function is used in multiple files,  I would keep it in the file where it is used.

Reply:
I tried to separate more "general" functions which might be used anywhere else into separate util files for 2 reasons:
1. A function might be used later in different modules
2. Provides better modularization. `Item` is focused more on its things `type`, `data` etc. It is similar for `ListenerManager` - when we remove `indexOfListener` into a util then `ListenerManager` can focus more on `register` `unregister` etc and not on data processing.
3. Allows testing not only the full data layer or specific modules but as well very basic functions and that may be useful. Sometimes it is hard to find why some tests are failing because they are quite general. If we test as well little "atoms" from code it helps to find the reason)
WDYT @jckautzmann

Reply:
IMHO utils make sense if the method is used in several places.
What do others think? 
@bpauli @vladbailescu

Reply:
I like the modular approach as it simplifies code scope and allows easier testing.